html/janus.js

```diff
@@ -1445,12 +1453,17 @@ function Janus(gatewayCallbacks) {
 						return false;
 					}
 
-					getUserMedia(
-						{audio: audioExist && isAudioSendEnabled(media), video: videoExist ? videoSupport : false},
-						function(stream) { pluginHandle.consentDialog(false); streamsDone(handleId, jsep, media, callbacks, stream); },
-						function(error) { pluginHandle.consentDialog(false); callbacks.error(error); });
+					navigator.mediaDevices.getUserMedia({
+						audio: audioExist ? audioSupport : false,
+						video: videoExist ? videoSupport : false
+					})
+					.then(function(stream) { pluginHandle.consentDialog(false); streamsDone(handleId, jsep, media, callbacks, stream); })
+					.catch(function(error) { pluginHandle.consentDialog(false); callbacks.error(error); });
+				})
```

Comment:
Besides the afore mentioned navigator.mediaDevices point, is the promise-based getUserMedia supported by ALL browser versions around? We didn't start using it yet as there may be versions of Chrome and Firefox who don't support it that we still want to allow to use Janus.